### ISSUE TITLE
NaN loss after 6K steps with batch size 2048

### ISSUE THREAD

**Issue by viyjy** [mosaicml/diffusion#30]
Hi, I got NaN loss during the training of SD-2-base-256.yaml

**Comment by mvpatel2000** [comment]
Can you please provide more details, eg hyperparameters (are they the same as our config?), picture of loss curve, and system settings (number of GPUs, etc)?

**Comment by viyjy** [comment]
Hi, I use the same setting/hyperparameters as you provided in hydra-yamls/SD-2-base-256.yaml

The loss curve is shown below, I use 128 GPUs each with 40GB GPU memory.
![image](https://github.com/mosaicml/diffusion/assets/20442927/3d2200a1-a251-4727-824e-7800793b96a7)
BTW, when I use local machine, it can save checkpoints, but it won't if I use AWS batch.

**Comment by viyjy** [comment]
Update: I add gradient clipping to avoid the NaN loss

**Comment by Landanjs** [comment]
Apologies for the delay! Were you seeing consistent loss spikes with your training runs or were they transient? 

This may be due to numerical instabilities. We wanted to setup bfloat16 to increase stability, but the convolution kernels were poorly supported so it lead to significant slowdown. We heard there was increased support in torch 2, so we will try it again.

**Comment by viyjy** [comment]
I didn't observe NaN on my local machine (8 GPUs, batch size=18*8), even for more than 20K steps.
Since it's expensive to run a job with 128GPUs to get batch size 2048, I didn't try another run to see if I get the same NaN loss at 6K step.
